### PR TITLE
Renamed And Updated HSM Related Classes

### PR DESCRIPTION
**Pull request by kelvinator07**

### CODE REVIEW

**Review by marcos-iov — COMMENTED** [review]
I see there are multiple code smells reported by Sonar. Could you please create a story to take care of them?

File: src/main/java/co/rsk/federate/signing/hsm/client/HSMSigningClientRskMst.java [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 
 import co.rsk.federate.signing.hsm.message.SignerMessage;
-import co.rsk.federate.signing.hsm.message.SignerMessageVersion1;
+import co.rsk.federate.signing.hsm.message.SignerMessageV1;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import org.bouncycastle.util.encoders.Hex;
 
-public class HSMClientVersion2RskMst extends HSMClientVersion2 {
+public class HSMSigningClientRskMst extends PowHSMSigningClient {
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply src/main/java/co/rsk/federate/signing/hsm/client/HSMSigningClientRskMst.java]: Done

File: src/main/java/co/rsk/federate/signing/hsm/client/PowHSMSigningClientBTC.java [hunk 2]
```diff
@@ -16,20 +16,20 @@
 import java.util.Arrays;
 import java.util.List;
 
-public class HSMClientVersion2BTC extends HSMClientVersion2 implements HSMBookkeepingClient {
-    private final Logger logger = LoggerFactory.getLogger(HSMClientVersion2BTC.class);
+public class PowHSMSigningClientBTC extends PowHSMSigningClient implements HSMBookkeepingClient {
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply src/main/java/co/rsk/federate/signing/hsm/client/PowHSMSigningClientBTC.java]: Done

File: src/main/java/co/rsk/federate/signing/hsm/client/PowHSMSigningClientBTC.java [hunk 3]
```diff
@@ -16,20 +16,20 @@
...
-    public HSMClientVersion2BTC(HSMClientProtocol protocol, int version) {
+    public PowHSMSigningClientBTC(HSMClientProtocol protocol, int version) {
         super(protocol, version);
     }
 
     @Override
     protected final ObjectNode createObjectToSend(String keyId, SignerMessage message) {
         final String MESSAGE_FIELD = "message";
-        SignerMessageVersion2 messageVersion2 = (SignerMessageVersion2) message;
+        PowHSMSignerMessage messageVersion2 = (PowHSMSignerMessage) message;
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply src/main/java/co/rsk/federate/signing/hsm/client/PowHSMSigningClientBTC.java]: Done

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**